client.go

```diff
@@ -12,6 +12,7 @@ import (
 	"strings"
 	"time"
 
+	"github.com/ONSdigital/dp-api-clients-go/headers"
```

Comment:
I know we can use dependency manager to handle which version of an import we should be using but is it bad practice to have libraries which depend on each other?

Reply:
That is a good point. Being completely objective I don't think the `dp-rchttp` client should know about anything about headers. 
The role of the client is to send/retry failed request and that should be all it does. Having specific header knowledge here blurs the line of responsibility, breaks encapsulation and creates a coupled code. It should be up to the caller to prepare the request.

> Do only one thing but do it well.

Reply:
While I agree that this library shouldn't have any knowledge of headers, that is something we have been moving towards with the repo over time, but is a separate piece of work requiring far too much refactoring of all of our libraries and apps to be feasible at this time.

Furthermore, there isn't a cyclic dependency here since cyclic dependencies are between packages not modules.  It is certainly a code smell, but as this PR is not adding the cyclic dependency between modules (just moving it from `go-ns` to `dp-api-clients-go` it does not seem reasonable to block this work over.